src/main/java/com/google/devtools/build/lib/cmdline/Label.java

```diff
@@ -181,9 +179,17 @@ private static RepositoryName computeRepoNameWithRepoContext(
     if (parts.repo() == null) {
       // Certain package names when used without a "@" part are always absolutely in the main repo,
       // disregarding the current repo and repo mappings.
-      return ABSOLUTE_PACKAGE_NAMES.contains(parts.pkg())
-          ? RepositoryName.MAIN
-          : repoContext.currentRepo();
+      if (ABSOLUTE_PACKAGE_NAMES.contains(parts.pkg())) {
+        return RepositoryName.MAIN;
+      }
+      // The //external package can only be referenced by the main repo or by external repositories
+      // that don't use strict visibility. For the main module repoContext.currentRepo() is equal to
```

Comment:
IMO just saying "external repos defined in WORKSPACE" would be clearer. Maybe even say "The legacy //external package ...". Then it's more understandable why we're checking for strict visibility, because only WORKSPACE repos have non-strict visibility.